tests/generic_config_updater/test_eth_interface.py

```diff
@@ -0,0 +1,148 @@
+import ipaddress
+import logging
+import pytest
+import re
+
+from tests.common.helpers.assertions import pytest_assert
+from tests.generic_config_updater.gu_utils import apply_patch, expect_op_success, expect_op_failure
+from tests.generic_config_updater.gu_utils import generate_tmpfile, delete_tmpfile
+from tests.generic_config_updater.gu_utils import create_checkpoint, delete_checkpoint, rollback_or_reload
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture(autouse=True)
+def ensure_dut_readiness(duthost):
+    """
+    Setup/teardown fixture for each ipv6 test
+    rollback to check if it goes back to starting config
+
+    Args:
+        duthost: DUT host object under test
+    """
+
+    create_checkpoint(duthost)
+
+    yield
+
+    try:
+        logger.info("Rolled back to original checkpoint")
+        rollback_or_reload(duthost)
+    finally:
+        delete_checkpoint(duthost)
+
+
+def check_interface_status(duthost, field):
+    """
+    Returns current status for Ethernet0 of specified field
+
+    Args:
+        duthost: DUT host object under test
+        field: interface field under test
+    """
+
+    cmds = "show interface status Ethernet0"
+    output = duthost.shell(cmds)
+    pytest_assert(not output['rc'])
+    status_data = output["stdout_lines"]
+    
+    field_index = status_data[0].split().index(field)
+    status = status_data[2].split()[field_index]
```

Comment:
>2 [](http://example.com/codeflow?start=25&length=1)

The magic number 2 is sensitive to the output format. To make it stable, you can parse each line for Ethernet0 pattern.

Reply:
changed to parse for Ethernet0